src/frontend/hooks/useProjectInviteListener.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useRef } from "react";
+
+// Gregor TODO: Add method to API for (un)subscribing to project invites
+const subscribeToProjectInvites = (callback: (inviteKey: string) => void) => {
+  const intervalId = setInterval(() => callback("abc123"), 3000);
+  return () => clearInterval(intervalId);
+};
+
+export default function useProjectInviteListener(
+  onInviteReceived: (inviteKey: string) => void
+) {
+  const callbackRef = useRef(onInviteReceived);
+
+  useEffect(() => {
+    callbackRef.current = onInviteReceived;
+  }, [onInviteReceived]);
+
+  useEffect(() => {
+    const handleInvite = (key: string) => callbackRef.current(key);
+    const unsubscribe = subscribeToProjectInvites(handleInvite);
+
+    return () => unsubscribe();
+  }, []);
+}
```

Comment:
Does this pattern have any unintentional side effects? The way I would write this would be:

```js
export default function useProjectInviteListener(onInviteReceived) {
  useEffect(() => {
    const unsubscribe = subscribeToProjectInvites(onInviteReceived)
    return () => unsubscribe()
  },
  [onInviteReceived]
  )
}
```

However subscriptions are a bit of a pain with React I think, and I'm unsure about what is "safe", e.g. see https://gist.github.com/bvaughn/e25397f70e8c65b0ae0d7c90b731b189
I wonder if there is an existing module we can use for this to make sure we're not going to hit some weird edge-case

Reply:
To my knowledge it doesn't unless there's something super low-level. This subscription pattern is inspired by hooks similar to `useEventListener` [here](https://usehooks.com/useEventListener/) where using a ref for the callback prevents potentially over (un)subscribing listeners. not adamant keeping it this way and fine with going with the more traditional solution